libraries/aws/jobs/src/aws_iot_jobs_serialize.c

```diff
@@ -1008,106 +1009,102 @@ static AwsIotJobsError_t _parseErrorDocument( const char * pErrorDocument,
                                    CODE_KEY,
                                    CODE_KEY_LENGTH,
                                    &pCode,
-                                   &codeLength ) == false )
+                                   &codeLength ) == true )
     {
-        IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_JOBS_BAD_RESPONSE );
-    }
-
-    /* Match the JSON error code to a Jobs return value. Assume invalid status
-     * unless matched.*/
-    status = AWS_IOT_JOBS_BAD_RESPONSE;
+        switch( codeLength )
+        {
+            /* InvalidJson */
+            case 13:
```

Comment:
These should be #defines instead of magic numbers.

Reply:
+1 Although this PR is not addressing updating  code other than gotos, it should be easy fix.

Reply:
I thought they are error code but actually not. They are the string length. For example, 13 is for "InvalidJson" (including quote).

The way it does is not elegant. I think ideally a "dictionary" data structure is perfect for such case, which can be implemented by a Trie. For the purpose of this PR, I suggest we leave it.

Reply:
Not easy fix because what it does here is to match a string to a set of per-defined ones.